packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -39,10 +43,14 @@ class CodeView extends StatefulWidget {
   final void Function(ScriptRef scriptRef, int line) onSelected;
 
   @override
-  _CodeViewState createState() => _CodeViewState();
+  _CodeViewState createState() => _CodeViewState(controller);
 }
 
 class _CodeViewState extends State<CodeView> with AutoDisposeMixin {
+  _CodeViewState(this.debuggerController);
+
+  final DebuggerController debuggerController;
```

Comment:
access controller via widget.controller instead of passing it into the state object

Reply:
And actually, since CodeViewState is a stateful widget, we can access the controller from Provider in didChangeDependencies. See https://github.com/flutter/devtools/blob/4456f8124ff2a553f0207ed3010ab663b7c23b16/packages/devtools_app/lib/src/debugger/debugger_screen.dart#L91 for an example of that

Reply:
why not just lookup the Provider in the build method?